### PR TITLE
Feat/civ 239

### PR DESCRIPTION
**Pull request by bhssamray**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-239

### Change description ###
This PR includes the changes required to capture debt details in LR spec journey along with some refactor upon based brach of ticket ROC-9783

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: ccd-definition/CaseEventToFields/DefendantResponseAdmitPartLRspec.json [hunk 1]
```diff
@@ -101,5 +101,59 @@
...
+    "ShowSummaryChangeOption": "Y"
+  },
+  {
+    "CaseTypeID": "CIVIL",
+    "CaseEventID": "DEFENDANT_RESPONSE_SPEC",
+    "CaseFieldID": "specDefendant1Debts",
+    "PageID": "DefendantDebts",
+    "PageDisplayOrder": 12,
+    "PageFieldDisplayOrder": 2,
+    "DisplayContext": "COMPLEX",
```
Comment (bhssamray) [inline ccd-definition/CaseEventToFields/DefendantResponseAdmitPartLRspec.json:155]: Change the DisplayContext to Mandatory and see if it works.

File: ccd-definition/CaseEventToComplexTypes/DefendantResponseLRspec.json [hunk 2]
```diff
@@ -48,6 +48,94 @@
     "DisplayContext": "OPTIONAL",
     "FieldShowCondition": "respondent1=\"DO_NOT_SHOW_IN_UI\""
   },
+  {
+    "ID": "Debts",
+    "CaseEventID": "DEFENDANT_RESPONSE_SPEC",
```
Comment (bhssamray) [inline ccd-definition/CaseEventToComplexTypes/DefendantResponseLRspec.json:53]: If DisplayContext MANDATORY works, please remove these entries from file

File: ccd-definition/CaseEventToFields/DefendantResponseAdmitPartLRspec.json [hunk 3]
```diff
@@ -101,5 +101,59 @@
...
+    "ShowSummaryChangeOption": "Y",
+    "CallBackURLMidEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/cases/callbacks/mid/validate-payment-date"
```
Comment (bhssamray) [inline ccd-definition/CaseEventToFields/DefendantResponseAdmitPartLRspec.json:157]: Need to remove this callback

File: ccd-definition/CaseEventToComplexTypes/DefendantResponseLRspec.json (hunk 2 above)
Comment (bhssamray) [reply ccd-definition/CaseEventToComplexTypes/DefendantResponseLRspec.json:53]: Removed these entries after use of MANDATORY DisplayContext in CaseEventToFields/DefendantResponseAdmitPartLRspec.json

File: ccd-definition/CaseEventToFields/DefendantResponseAdmitPartLRspec.json (hunk 1 above)
Comment (bhssamray) [reply ccd-definition/CaseEventToFields/DefendantResponseAdmitPartLRspec.json:155]: Changed DisplayContext to MANDATORY

File: ccd-definition/CaseEventToFields/DefendantResponseAdmitPartLRspec.json (hunk 3 above)
Comment (bhssamray) [reply ccd-definition/CaseEventToFields/DefendantResponseAdmitPartLRspec.json:157]: Removed callback as not needed here. was added for testing in dev local

**Review by sampankumar — APPROVED** [review]

**Review by indu2019 — APPROVED** [review]
approved

**Pull request merged**